runtime/hsa-runtime/core/runtime/amd_blit_kernel.cpp

```diff
@@ -1272,7 +1272,7 @@ void BlitKernel::PopulateQueue(uint64_t index, uint64_t code_handle, void* args,
   std::atomic_thread_fence(std::memory_order_acquire);
   queue_buffer[index & queue_bitmask_] = packet;
   std::atomic_thread_fence(std::memory_order_release);
-  if (core::Runtime::runtime_singleton_->flag().dev_mem_queue() && !queue_->needsPcieOrdering()) {
+  if (queue_->IsDeviceMem() && !queue_->needsPcieOrdering()) {
```

Comment:
@saleelk I'm curious if the logic is correct for the original needsPcieOrdering() method you added. Shouldn't this really be:

`queue_->needsPcieOrdering()`

Meaning we need to change the logic of that call internally?